implants/lib/eldritch/src/process/name_impl.rs

```diff
@@ -1,5 +1,67 @@
 use anyhow::Result;
+use sysinfo::{Pid, PidExt, ProcessExt, System, SystemExt};
 
 pub fn name(_pid: i32) -> Result<String> {
-    unimplemented!("Method unimplemented")
-}
\ No newline at end of file
+    if !System::IS_SUPPORTED {
+        return Err(anyhow::anyhow!("This OS isn't supported for process functions.
+         Pleases see sysinfo docs for a full list of supported systems.
+         https://docs.rs/sysinfo/0.23.5/sysinfo/index.html#supported-oses\n\n"));
+    }
+
+    let mut sys = System::new();
+    sys.refresh_processes();
+
+    let mut res = "";
+
+    if let Some(process) = sys.process(Pid::from_u32(_pid as u32)) {
+        res = process.name()
+    }
+    
+    Ok(res.to_string())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{process::Command};
+
+    #[test]
+    fn test_process_name() -> anyhow::Result<()>{
+        let mut commandstring = "sleep 5";
+        if cfg!(target_os = "linux") || 
+                cfg!(target_os = "ios") || 
+                cfg!(target_os = "macos") || 
+                cfg!(target_os = "android") || 
+                cfg!(target_os = "freebsd") || 
+                cfg!(target_os = "openbsd") ||
+                cfg!(target_os = "netbsd") {
+            commandstring = "sleep";
+        } else if cfg!(target_os = "windows") {
+            commandstring = "timeout";
+        }
```

Comment:
Could you add an else here to make it clear the test will fail on non-supported OSes.
```rust
else {
    return anyhow::anyhow!(Err("OS Not supported please re run on Linux, Windows, or MacOS");
}
```